resources/assets/lib/stores/channel-store.ts

```diff
@@ -232,12 +232,16 @@ export default class ChannelStore implements DispatchListener {
   }
 
   @action
-  private handleChatMessageNewEvent(event: MessageNewEvent) {
+  private async handleChatMessageNewEvent(event: MessageNewEvent) {
     for (const message of event.json.messages) {
       const channel = this.channels.get(message.channel_id);
-      if (channel == null) continue;
 
-      channel.addMessage(message);
+      if (channel != null) {
+        channel.addMessage(message);
+      } else if ((event.json.type === 'PM' || event.json.type === 'ANNOUNCE')) {
```

Comment:
extra parentheses aside, with `getChannel` returning 403 for unjoined public channel, the type check can probably be skipped (and check status code instead)? This will also allow the channel to appear if the user joined it in game without having to refresh the page

Reply:
The check is to avoid making multiple `getChannel` requests when receiving messages from public channels after leaving them

Reply:
That sounds like a same problem if multiple pm/announcement messages arrive after leaving them 🤔